packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue

```diff
@@ -63,7 +63,40 @@
         </li>
       </oc-list>
       <div v-if="links.length > 3" class="oc-flex oc-flex-center">
-        <oc-button appearance="raw" @click="toggleLinkListCollapsed" v-text="collapseButtonTitle" />
+        <oc-button appearance="raw" @click="toggleLinkListCollapsed">
+          <span v-text="collapseButtonTitle" />
+          <oc-icon :name="collapseButtonIcon" fill-type="line" />
+        </oc-button>
+      </div>
+      <div v-if="indirectLinks.length">
+        <hr class="link-separator oc-my-m" />
+        <h3 class="oc-text-bold oc-m-rm oc-text-initial">
+          <span v-text="indirectLinksHeading" />
+          <oc-contextual-helper v-if="helpersEnabled" v-bind="indirectLinkHelp" />
+        </h3>
+        <oc-list v-if="displayIndirectLinks.length" class="oc-overflow-hidden oc-my-m">
+          <li
+            v-for="link in displayIndirectLinks"
+            :key="link.key"
+            class="oc-py-s"
+            :data-testid="`files-link-id-${link.id}`"
+          >
+            <name-and-copy :link="link" />
+            <details-and-edit
+              :available-role-options="availableRoleOptions"
+              :expiration-date="expirationDate"
+              :is-folder-share="highlightedFile.isFolder"
+              :is-modifiable="false"
+              :link="link"
+            />
+          </li>
+        </oc-list>
+        <div v-if="indirectLinks.length > 3" class="oc-flex oc-flex-center">
```

Comment:
Indirect link collapse/expand toggle is about `show 0` vs `show all`, no? I think you can get rid of the `v-if="indirectLinks.length > 3"` check entirely.

Reply:
Taken care of, thx